internal/controller/scheduler/scheduler.go

```diff
@@ -586,6 +586,18 @@ func (w *worker) Build(podSpec *corev1.PodSpec, skipCheckout bool, inputs buildI
 		w.logger.Debug("Applied podSpec patch from k8s plugin", zap.Any("patched", patched))
 	}
 
+	if skipCheckout {
+		for i, pod := range podSpec.Containers {
+			switch pod.Name {
+			case CheckoutContainerName:
+				podSpec.Containers = append(podSpec.Containers[:i], podSpec.Containers[i+1:]...)
```

Comment:
[slices.Delete](https://pkg.go.dev/slices#Delete) would be more readable here:

```suggestion
				podSpec.Containers = slices.Delete(podSpec.Containers, i, i+1)
```